File: Programming/Lv.1/최소 직사각형/solution.py
# https://school.programmers.co.kr/learn/courses/30/lessons/86491?language=python3
def solution(sizes):
    sizes = [[min(s[0], s[1]), max(s[0], s[1])] for s in sizes]
    return max([s[0] for s in sizes]) * max([s[1] for s in sizes])

File: Programming/Lv.1/최소 직사각형/solution.js
// https://school.programmers.co.kr/learn/courses/30/lessons/86491
function solution(sizes) {
  return sizes
    .map((v) => (v[0] < v[1] ? [v[1], v[0]] : v))
    .reduce(
      (p, cv) => {
        p[0] = p[0] < cv[0] ? cv[0] : p[0];
        p[1] = p[1] < cv[1] ? cv[1] : p[1];
        return p;
      },
      [0, 0]
    )
    .reduce((p, cv) => p * cv, 1);
}
